packages/client/src/SatisfiesSemver.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2024 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type {
+  GreaterThan,
+  GreaterThanOrEqual,
+  IsEqual,
+  LessThan,
+} from "type-fest";
+
+export type VersionString<
```

Comment:
Does this type need to be exported here and in api?

Reply:
I didn't want to make it part of our API contract, thus its existence twice.